### PR TITLE
Enable floating promises check everywhere (fix issues)

### LINKED ISSUE #1080: possible bug in faucet
https://github.com/celo-org/celo-monorepo/blob/8dacf1eb43a914c8d3b612936ad73ed727d0417d/packages/faucet/src/database-helper.ts#L260

First Thanks for open sourcing everything.
While browsing to check how you solved similar challenges we have at GoodDollar, the above line of code seems incorrect to me as trySetLock returns a promise so using it in an if condition would always return true. It should probably be ``&& (await trySetLock)`` or am I missing something?

### PR DESCRIPTION
**Pull request by mcortesi**
### Description

Removes floagint promises on all packages that inherit from tslint
parent config.

Only in `mobile` package, the config is overrided as a "warning".

Also: updates tslint & tslint rule set to their last versions.

### Tested

`yarn lint`

### Other changes

Upgrade tslint

### Related issues

- Related to #238 
- Fixes #1080

### CODE REVIEW

File: packages/contractkit/src/address-registry.ts [hunk 1]
```diff
@@ -22,8 +21,8 @@ export class AddressRegistry {
   async addressFor(contract: CeloContract): Promise<Address> {
     if (!this.cache.has(contract)) {
       debug('Fetching address from Registry for %s', contract)
-      const hash = Web3.utils.soliditySha3({ type: 'string', value: contract })
-      const address = await this.registry.methods.getAddressFor(hash).call()
+      // const hash = Web3.utils.soliditySha3({ type: 'string', value: contract })
```
Comment (cmcewen) [inline packages/contractkit/src/address-registry.ts]: why comment this out?

File: packages/verifier/src/components/HomeScreen/HomeScreen.tsx [hunk 2]
```diff
@@ -86,20 +86,24 @@ class HomeScreen extends React.Component<Props, State> {
   }
 
   async componentDidMount() {
-    this.getVerifierStatus()
-    NetInfo.addEventListener('connectionChange', this.handleNetworkStatusChange)
+    try {
+      await this.getVerifierStatus()
+      NetInfo.addEventListener('connectionChange', this.handleNetworkStatusChange)
```
Comment (cmcewen) [inline packages/verifier/src/components/HomeScreen/HomeScreen.tsx:90]: I think this should probably go above anything we await

File: packages/verifier/src/components/HomeScreen/HomeScreen.tsx [hunk 3]
```diff
@@ -86,20 +86,24 @@ class HomeScreen extends React.Component<Props, State> {
...
+       * Because we are getting this from our custom VerifierService, there is no way to subscribe to updates
+       * To ensure we keep it up to date, we will retrieve it every time the home page mounts
+       * To avoid this, we would need to add the RN Firebase module for messaging and subscribe to its updates
+       */
+      const fcmToken = await VerifierService.getFCMToken()
+      await setFcmToken(fcmToken)
 
-    if (!this.props.isVerifying && isVerifyingDisabledLongTime(this.props.verifyingOffAt)) {
-      setTimeout(() => this.showVerifyingOffLongMessage(), 2000)
+      if (!this.props.isVerifying && isVerifyingDisabledLongTime(this.props.verifyingOffAt)) {
```
Comment (cmcewen) [inline packages/verifier/src/components/HomeScreen/HomeScreen.tsx]: also move this above await probably

File: packages/contractkit/src/address-registry.ts (hunk 1 above)
Comment (mcortesi) [reply packages/contractkit/src/address-registry.ts]: Thanks for catching this.. my commit included some other work that should not be here.

Now that's fixed

**Review by cmcewen — APPROVED** [review]

**Review by jmrossy — APPROVED** [review]
Very nice!

**Pull request merged**